pkg/cmd/pomerium/pomerium.go

```diff
@@ -94,6 +96,8 @@ func New(opts ...Option) *Pomerium {
 }
 
 func (p *Pomerium) Start(ctx context.Context, tracerProvider oteltrace.TracerProvider, src config.Source) error {
+	p.startMu.Lock()
+	defer p.startMu.Unlock()
 	updateTraceClient(ctx, src.GetConfig())
```

Comment:
is that mutex for tests? maybe we can move it to testenv then?

Reply:
This is to synchronize start/stop operations, for example if you start pomerium and then quickly stop it, some things might not be initialized.